### ISSUE TITLE
pruning yolov8 model

### ISSUE THREAD

**Issue by Christian-lyc** [ultralytics/ultralytics#16395]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, sorry, I have used pytorch l1 unstructured pruning function to prune the model.
 I'm not sure how you calculate numbers of parameters with training function, it seems after pruning, when I finetune the model, it gives same numbers of parameters. This ''Model summary (fused)'' shows same parameters. What I thought is pruned model should be smaller?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Christian-lyc, thank you for reaching out and for your interest in Ultralytics 🚀!

It sounds like you're diving into model pruning, which is an exciting area. If you haven't already, please check out our [Docs](https://docs.ultralytics.com) for detailed usage guidelines and existing discussions around model optimization.

If this is indeed a 🐛 Bug Report, ensure a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) is provided to help us understand and debug the issue.

For custom training or model pruning ❓ Questions, make sure to include as much detail as possible, like your pruning approach specifics, dataset examples, and training logs. Ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To stay connected with the Ultralytics community:
- Join our real-time chat on [Discord](https://ultralytics.com/discord) 🎧.
- Participate in discussions on [Discourse](https://community.ultralytics.com).
- Explore our [Subreddit](https://reddit.com/r/ultralytics) for diverse threads and knowledge sharing.

## Upgrade

Make sure you've upgraded to the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLOv8 in various environments. Choose the best one for you:

- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Check our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Use our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Check our CI status here: <a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>. If the badge is green, all tests pass, ensuring correct operation across various platforms.

This is an automated response, but rest assured, an Ultralytics engineer will review your issue soon. We appreciate your patience and collaboration! 😊

**Comment by Christian-lyc** [comment]
So sorry, I realized later. It was stupid

**Comment by Christian-lyc** [comment]
So sorry, I realized later. It was stupid question.

**Comment by glenn-jocher** [comment]
No worries at all! Feel free to reach out if you have any more questions.

**Comment by Christian-lyc** [comment]
Hi, I have another question about structure pruning. I used pytorch structure pruning for global pruning. I could find 0 in weight_mask, before prune.remove(module, 'weight')
My code is like this:
for name, module in model.model.named_modules():
if isinstance(module, nn.Conv2d):
if 0 in module.weight_mask:
print(module.weight)
prune.remove(module, 'weight')

print(module.weight) can show 0 weights, but after prune.remove, it still shows that the weight as original. Do you know what problem is there? Thank you so much

**Comment by Christian-lyc** [comment]
Sorry, I checked again, after remove function, the pruned part still shows 0, which should be removed.
